contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--2.3.0--2.4.0.sql

```diff
@@ -817,6 +817,16 @@ and
   );
 GRANT SELECT ON sys.types TO PUBLIC;
 
+-- function sys.object_id(object_name, object_type) needs to change input type to VARCHAR
+ALTER FUNCTION sys.object_id RENAME TO object_id_deprecated_in_2_4_0;
```

Comment:
what is out plan for 3.0 to 3.1? we can do the same thing there but that will result in unnecessary rename and user has to again recreate dependent views. May be we detect that existence of deprecated object and skip recreate in 3.0 to 3.1

Reply:
Rishabh said that `We can check if column datatype of sys.object_id is already varchar , if yes then we skip drop+recreate step`.